src/components/forms/NumberInputField.tsx

```diff
@@ -148,7 +128,35 @@ export function NumberInputField(props: Props) {
                         ? reactSettings.number_precision
                         : 2,
                   }).value
-                )
+                : undefined;
+
+              setCurrentValue(formattedValue);
+            }
+          }}
+          onBlur={(event) => {
+            if (props.onValueChange && !props.changeOverride) {
+              props.onValueChange(
+                event.target.value
+                  ? String(
+                      currency(event.target.value, {
+                        separator: company?.use_comma_as_decimal_place
+                          ? '.'
+                          : ',',
+                        decimal: company?.use_comma_as_decimal_place
+                          ? ','
+                          : '.',
+                        symbol: '',
+                        precision:
+                          typeof props.precision === 'number'
+                            ? props.precision
+                            : reactSettings?.number_precision &&
+                              reactSettings?.number_precision > 0 &&
+                              reactSettings?.number_precision <= 100
+                            ? reactSettings.number_precision
+                            : 2,
+                      }).value
```

Comment:
This is really starting to get out of hand. Can we extract this into separate function with clean if statements?

Reply:
@beganovich Sure thing, the logic has been separated into a function and then applied to the logic. Let me know your thoughts.